### ISSUE TITLE
Error using dashboard filter: Function "ID" not found

### ISSUE THREAD

**Issue by jabbett** [metabase/metabase#10084]
For the first time, I've attempted to create a dashboard with a filter.

With an empty filter, everything looks fine:

![image](https://user-images.githubusercontent.com/923971/58831293-61d2fe00-861a-11e9-8fc1-69ab7adece1a.png)

When I search for and select something to filter by, I get a generic error:

![image](https://user-images.githubusercontent.com/923971/58831326-72837400-861a-11e9-9794-56d0bbeedb0c.png)

In the logs, the error is this:

```
Jun 03 20:05:27 WARN metabase.query-processor :: Query failure: Function "ID" not found; SQL statement:
SELECT "METABASE_FIELD"."BASE_TYPE" FROM "METABASE_FIELD" WHERE field-id("ID", 1006) LIMIT 1 [90022-197]
("clojure.core$ex_info.invokeStatic(core.clj:4739)"
"clojure.core$ex_info.invoke(core.clj:4739)"
"--> query_processor$assert_query_status_successful.invokeStatic(query_processor.clj:285)"
"query_processor$assert_query_status_successful.invoke(query_processor.clj:277)"
"query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:323)"
"query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:316)"
"query_processor$fn__37661$process_query_and_save_execution_BANG___37666$fn__37667.invoke(query_processor.clj:354)"
"query_processor$fn__37661$process_query_and_save_execution_BANG___37666.invoke(query_processor.clj:340)"
"api.card$run_query_for_card.invokeStatic(card.clj:580)"
"api.card$run_query_for_card.doInvoke(card.clj:566)"
"api.card$fn__46963$fn__46966.invoke(card.clj:587)"
"api.card$fn__46963.invokeStatic(card.clj:586)"
"api.card$fn__46963.invoke(card.clj:582)"
"middleware$enforce_authentication$fn__56091.invoke(middleware.clj:113)"
"api.routes$fn__56237.invokeStatic(routes.clj:62)"
"api.routes$fn__56237.invoke(routes.clj:62)"
"routes$fn__56326$fn__56327.doInvoke(routes.clj:108)"
"routes$fn__56326.invokeStatic(routes.clj:103)"
"routes$fn__56326.invoke(routes.clj:103)"
"middleware$catch_api_exceptions$fn__56226.invoke(middleware.clj:436)"
"middleware$log_api_call$fn__56204$fn__56206.invoke(middleware.clj:364)"
"middleware$log_api_call$fn__56204.invoke(middleware.clj:363)"
"middleware$add_security_headers$fn__56146.invoke(middleware.clj:252)"
"core$wrap_streamed_json_response$fn__62590.invoke(core.clj:67)"
"middleware$bind_current_user$fn__56096.invoke(middleware.clj:137)"
"middleware$maybe_set_site_url$fn__56156.invoke(middleware.clj:290)"
"middleware$add_content_type$fn__56149.invoke(middleware.clj:262)")
```

No matter the filter value I choose, I see the same error in the log.

- My browser: Chrome 74.0.3729.169 (64-bit)
- My OS: OS X 10.12.6
- Database: Postgres
- Metabase version: 0.31.1
- Metabase hosting environment: Docker container on Linux
- Metabase internal database: H2

**Comment by flamber** [comment]
Hi @jabbett 
I'm pretty sure this was fixed in #9025, which was part of 0.31.2. The newest release is 0.32.8.
Always remember to backup before upgrading!
And please be careful with using H2 in production - it's recommended that you migrate:
https://metabase.com/docs/latest/operations-guide/start.html#migrating-from-using-the-h2-database-to-mysql-or-postgres

**Comment by salsakran** [comment]
@jabbett was the thumbs up to reflect that upgrading fixed this?

**Comment by jabbett** [comment]
Hi, @salsakran -- that was just a thankful acknowledgement of the advice ;)

I've submitted a ticket to my infrastructure team to have our Metabase server upgraded. Will post back as soon as that happens.

**Comment by salsakran** [comment]
closing since I think this was fixed in #9025 and there have been no further reports.

**Comment by jabbett** [comment]
(Just to close the loop - yes, the upgrade the solved the problem.)
